numpy/core/tests/test_multiarray.py

```diff
@@ -6276,6 +6276,13 @@ def __mul__(self, other):
         with assert_raises(TypeError):
             b = np.matmul(a, a)
 
+    def test_matmul_bool(self):
+        # gh-14439
+        a = np.array([[1, 0],[1, 1]], dtype=bool)
+        assert np.max(a.view(np.int8)) == 1
+        b = np.matmul(a, a)
+        # matmul with boolean output should always be 0, 1
+        assert np.max(b.view(np.int8)) == 1
```

Comment:
Could be nice to have a larger than 2x2 array as well I guess. And while we are at it, ensure:
```
np.matmul(np.empty((2, 0), dtype=np.bool_), np.empty(0, dtype=np.bool_))
```
gives `[False, False]` (that is what dot gives, I assume it is correct, heh).

Reply:
Should probably view as `uint8` here, since currently this test won't detect `-1`

Reply:
fixed `uint8`, added more tests. `np.empty` gives uninitialized memory, used `np.zeros` instead.